Models/Shape.cs

```diff
@@ -21,33 +49,86 @@ public class PolylineShape : ShapeBase
     public double StrokeThickness { get; set; } = 1.0;
     public List<Point> Points { get; set; } = new();
     
-    public override Control ToControl() 
+    protected override Rectangle CreateBoundingBoxVisual()
     {
-        return new Polyline
+        var rect = base.CreateBoundingBoxVisual();
+        rect.Stroke = Brushes.Red;
+        rect.StrokeThickness = 1;
+        return rect;
+    }
+    
+    public override Control ToControl()
+    {
+        var canvas = (Canvas)base.ToControl();
+
+        var polyline = new Polyline
         {
-                Points = new AvaloniaList<Point>(Points),
-                Stroke = Brushes.Black,
-                StrokeThickness = StrokeThickness
-       };
+            Points = new AvaloniaList<Point>(Points),
+            Stroke = Brushes.Black,
+            StrokeThickness = StrokeThickness
+        };
+
+        canvas.Children.Add(polyline);
+        return canvas;
+    }
+    public void AddPoint(Point next)
+    {
+        Points.Add(next);
+        double left   = Math.Min(BoundingBox.X, next.X);
+        double top    = Math.Min(BoundingBox.Y, next.Y);
+        double right  = Math.Max(BoundingBox.Right, next.X);
+        double bottom = Math.Max(BoundingBox.Bottom, next.Y);
+
+        BoundingBox = new Rect(left, top, right - left, bottom - top);
     }
 }
 
 public class GroupShape : ShapeBase
 {
     public List<ShapeBase> Children { get; set; } = new();
+    public void UpdateBoundingBox()
+    {
+        Rect? groupBox = null;
```

Comment:
I prefer [`Rect.Empty`](https://learn.microsoft.com/en-us/dotnet/api/system.windows.rect.empty?view=windowsdesktop-9.0#system-windows-rect-empty). Then `Union` should just work without any null check.